### ISSUE TITLE
App loses context (or restarts) when split screening on Android 7+

### ISSUE THREAD

**Issue by sitepodmatt** [m2049r/xmrwallet#156]
Using Android 7 or later with Monerujo 1.2.4 (45) wi

Steps:
Start monerujo
Open wallet or some other context
Open Android App Switcher
Open another app so the screen is now split screen (i.e. 50% monerujo, 50% another app)

Expect: nothing
Actual: 'Wrapping up'...  Monerujo seems to close and reopen, losing context

Other:
At first I thought maybe this is the expected behaving when switching apps, however I can switch away from Monerujo, to email or other app in normal full screen and then back to Monerujo without losing context, so I'm not sure what is special about split screening.

Device: Nokia 5, Stock vanilla Android 7.1.2

**Comment by m2049r** [comment]
Android App Switcher is a Nokia stock app?

**Comment by sitepodmatt** [comment]
Nokia have no skins or customizations. It's vanilla Android. Hail Nokia. 

https://www.greenbot.com/article/3105486/android/how-to-use-split-screen-mode-in-android-nougat.html (warning ad loaded site and autostart video ads)

**Comment by m2049r** [comment]
i think i know why this is - will have to look at it later.

**Comment by waninkoko** [comment]
The main difference is that, when splitting the screen, the activity/fragment is recreated, while switching between apps (normally) doesn't trigger this. This is probably the key about this issue.

**Comment by m2049r** [comment]
yes, the onCreate code in the wallet activity kills the activity if it is recreated to avoid having to deal with this. maybe we should deal with this. e.g. need to make sure the wallet service is running with the correct wallet.

https://github.com/m2049r/xmrwallet/blob/master/app/src/main/java/com/m2049r/xmrwallet/WalletActivity.java#L185-L190
